web-common/src/lib/analytics/posthog.ts

```diff
@@ -30,6 +30,24 @@ export function initPosthog(rillVersion: string, sessionId?: string | null) {
         "Rill version": rillVersion,
       });
     },
+    before_send: (event) => {
+      // Only filter $exception (error tracking) events by URL
+      if (!event) return null;
+      if (event.event === "$exception") {
+        const hostname = window.location.hostname;
+        if (
+          hostname === "ui.rilldata.com" ||
+          hostname === "localhost" ||
+          hostname === "127.0.0.1"
+        ) {
+          return event;
+        }
+        // Drop $exception events for all other URLs
+        return null;
+      }
+      // Allow all other events
+      return event;
```

Comment:
A little clean-up suggestion:
```suggestion
      if (event?.event !== "$exception") {
        return event;
      }

      // For $exception events, only send them from specific hostnames.
      // TODO: A ONE-LINE COMMENT THAT EXPLAINS WHY WE NEED TO DO THIS
      const hostname = window.location.hostname;
      const allowedHostnames = ["ui.rilldata.com", "localhost", "127.0.0.1"];
      if (allowedHostnames.includes(hostname)) {
        return event;
      }

      // Drop $exception events for all other hostnames.
      return null;
```

Reply:
Additionally, would you mind adding a comment that explains the motivation for this code? E.g. does this avoid collecting exceptions thrown by Chrome extensions?

Reply:
I meant a code comment. In the above snippet, I've added the "TODO" as a place for you to write the explanation :)